src/CSET/operators/ageofair.py

```diff
@@ -0,0 +1,489 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Age of air operator.
+
+This determines how old air is since it entered through the lateral boundary at some given leadtime.
+"""
+
+import datetime
+import logging
+import multiprocessing
+import os
+import tempfile
+from functools import partial
+from math import asin, cos, radians, sin, sqrt
+
+import numpy as np
+from iris.cube import Cube
+from scipy.ndimage import gaussian_filter
+
+from CSET.operators._utils import get_cube_yxcoordname
+
+
+def _calc_dist(coord_1, coord_2):
+    """Haversine distance in metres."""
+    # Approximate radius of earth in m
+    # Source: https://nssdc.gsfc.nasa.gov/planetary/factsheet/earthfact.html
+    radius = 6378000
+
+    # extract coordinates and convert to radians
+    lat1 = radians(coord_1[0])
+    lon1 = radians(coord_1[1])
+    lat2 = radians(coord_2[0])
+    lon2 = radians(coord_2[1])
+
+    # Find out delta latitude, longitude
+    dlon = lon2 - lon1
+    dlat = lat2 - lat1
+
+    # Compute distance
+    a = sin(dlat / 2) ** 2 + cos(lat1) * cos(lat2) * sin(dlon / 2) ** 2
+    c = 2 * asin(sqrt(a))
+    distance = radius * c
+
+    return distance
+
+
+def _aoa_core(
+    x_arr: np.ndarray,
+    y_arr: np.ndarray,
+    z_arr: np.ndarray,
+    g_arr: np.ndarray,
+    lats: np.ndarray,
+    lons: np.ndarray,
+    dt: int,
+    plev_idx: int,
+    timeunit: str,
+    cyclic: bool,
+    tmpdir: str,
+    lon_pnt: int,
+):
+    """AOA multiprocessing core.
+
+    Runs the core age of air code on a specific longitude point (all latitudes) for
+    parallelisation.
+
+    Arguments
+    ----------
+    x_arr: np.ndarray
+        A numpy array containing x wind data.
+    y_arr: np.ndarray
+        A numpy array containing y wind data.
+    Z_arr: np.ndarray
+        A numpy array containing w wind data, or a placeholder full of zeros if incW is False
+    g_arr: np.ndarray
+        A numpy array containing geopotential height data.
+    lats: np.ndarray
+        A numpy array containing latitude points.
+    lons: np.ndarray
+        A numpy array containing longitude points.
+    dt: int
+        Gap between time intervals
+    plev_idx: int
+        Index of pressure level requested to run back trajectories on.
+    timeunit: str
+        Units of time, currently only accepts 'hour'
+    cyclic: bool
+        Whether to wrap at east/west boundaries. See compute_ageofair for a fuller description.
+    tmpdir: str
+        Path to store intermediate data
+    lon_pnt: int
+        Longitude point to extract and run back trajectories on for parallelisation.
+    """
+    # Initialise empty array to store age of air for this latitude strip.
+    ageofair_local = np.zeros((x_arr.shape[0], x_arr.shape[2]))
+    logging.info("Working on %s", lon_pnt)
```

Comment:
```suggestion
    logging.debug("Working on %s", lon_pnt)
```
This is more debugging information.